microk8s-resources/actions/traefik.yaml

```diff
@@ -0,0 +1,152 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: traefik-ingress-controller
+  namespace: traefik
+---
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: traefik-ingress-controller
+  namespace: traefik
+  labels:
+    k8s-app: traefik-ingress-lb
+spec:
+  selector:
+    matchLabels:
+      k8s-app: traefik-ingress-lb
+      name: traefik-ingress-lb
+  template:
+    metadata:
+      labels:
+        k8s-app: traefik-ingress-lb
+        name: traefik-ingress-lb
+    spec:
+      tolerations:
+      - effect: NoSchedule
+        operator: Exists
+      serviceAccountName: traefik-ingress-controller
+      terminationGracePeriodSeconds: 60
+      hostNetwork: true
+      containers:
+      - image: traefik:latest
+        name: traefik-ingress-lb
+        ports:
+        - name: http
+          containerPort: 8080
+          hostPort: 8080
+          # - name: admin
+          # containerPort: 8090
+          #hostPort: 8090
+        securityContext:
+          capabilities:
+            drop:
+            - ALL
+            add:
+            - NET_BIND_SERVICE
+        args:
+        - --providers.kubernetesingress=true
+        # you need to manually set this IP to the incoming public IP
+        # that your ingress resources would use. Note it only affects
+        # status and kubectl UI, and doesn't really do anything
+        # It could even be left out https://github.com/containous/traefik/issues/6303
+        - --providers.kubernetesingress.ingressendpoint.ip=127.0.0.1
+        ## uncomment these and the ports above and below to enable
+        ## the web UI on the host NIC port 8080 in **insecure** mode
+        # - --api.dashboard=true
+        # - --api.insecure=true
```

Comment:
Uncommenting these lines to enable the dashboard causes the pod to crash with a port conflict:

```
traefik.go:76: command traefik error: error while building entryPoint traefik: error preparing server: error opening listener: listen tcp :8080: bind: address already in use
```

Also, shouldn't the CustomResourceDefinitions have been included in this service, or is there a reason to not use them?

Reply:
Hi
enabling this parameters for dashboard is insecure, hence by default its disabled, yes CRD to this service is the right way to enable dashboard which available recently . traefik version used here is 2.23, if user wants dashboard to know about services/pods/containers they can use portainer.